table/block_based/filter_policy.cc

```diff
@@ -443,7 +443,8 @@ void BloomFilterPolicy::CreateFilter(const Slice* keys, int n,
                                      std::string* dst) const {
   // We should ideally only be using this deprecated interface for
   // appropriately constructed BloomFilterPolicy
-  assert(mode_ == kDeprecatedBlock);
+  // FIXME disabled because of bug in C interface; see issue #6129
+  //assert(mode_ == kDeprecatedBlock);
```

Comment:
Does it make sense to use `assert(mode_ == kDeprecatedBlock || mode_ == kAuto)`?

Reply:
No, you have to explicitly opt in to using the block-based filter format, which we call "deprecated" in some places

Reply:
In fact, we would not have caught the bug if the assertion were so permissive.

Reply:
I think the long term solution is to fix the C-API related code. At the moment as a temporary fix, adding an `or` seems stricter than removing the whole assertion.